src/WidgetBase.ts

```diff
@@ -272,10 +262,38 @@ export class WidgetBase<P = WidgetProperties, C extends DNode = DNode> extends E
 		vnodeSelector: string,
 		properties: VNodeProperties
 	): void {
-		this._setNode(element, properties);
+		this._nodeHandler.add(element, properties);
 		this.onElementUpdated(element, String(properties.key));
 	}
 
+	protected _afterRootUpdateCallback(
+		element: Element,
+		projectionOptions: ProjectionOptions,
+		vnodeSelector: string,
+		properties: VNodeProperties
+	): void {
+		this._addElementToNodeHandler(element, projectionOptions, properties);
+		this.onElementUpdated(element, String(properties.key));
+	}
+
+	_addElementToNodeHandler(element: Element, projectionOptions: ProjectionOptions, properties: VNodeProperties) {
+		this._currentRootNode += 1;
+
+		if (!this._projectorAttachEvent) {
```

Comment:
minor nit: I believe `if (this._projectorAttachEvent === undefined) {` is more performant and also personally I find it clearer.